common/cuda_hip/matrix/batch_ell_kernels.hpp.inc

```diff
@@ -165,3 +165,43 @@ __global__ void scale_kernel(
         scale(col_scale_b, row_scale_b, mat_b);
     }
 }
+
+
+template <typename ValueType, typename IndexType>
+__device__ __forceinline__ void add_scaled_identity(
+    const ValueType alpha, const ValueType beta,
+    const gko::batch::matrix::ell::batch_item<ValueType, IndexType>& mat)
+{
+    for (int tidx = threadIdx.x; tidx < mat.num_rows; tidx += blockDim.x) {
+        for (size_type idx = 0; idx < mat.num_stored_elems_per_row; idx++) {
+            const auto ind = tidx + idx * mat.stride;
+            mat.values[ind] *= beta;
+            const auto col_idx = mat.col_idxs[ind];
+            if (col_idx == invalid_index<IndexType>()) {
+                break;
+            } else {
+                if (tidx == col_idx) {
+                    mat.values[ind] += alpha;
+                }
+            }
```

Comment:
nit: you can combine them to `else if`

Reply:
IMO this is easier to read. The compiler should combine them so you should not lose any performance.